File: examples/Fan_PRB_65_2002/fig12.lua
-- Bottom pane of Fig. 12 in
-- Shanhui Fan and J. D. Joannopoulos,
-- "Analysis of guided resonances in photonic crystal slabs",
-- Phys. Rev. B, Vol. 65, 235112

S = S4.NewSimulation()
S:SetLattice({1,0}, {0,1})
S:SetNumG(100)
S:AddMaterial("Silicon", {12,0}) -- real and imag parts
S:AddMaterial("Vacuum", {1,0})

S:AddLayer('AirAbove', 0 , 'Vacuum')
S:AddLayer('Slab', 0.5, 'Silicon')
S:SetLayerPatternCircle('Slab', 'Vacuum', {0,0}, 0.2)
S:AddLayerCopy('AirBelow', 0, 'AirAbove')

S:SetExcitationPlanewave(
	{0,0}, -- incidence angles
	{1,0}, -- s-polarization amplitude and phase (in degrees)
	{0,0}) -- p-polarization amplitude and phase

--S:UsePolarizationDecomposition()

for freq=0.25,0.27,0.003 do
	S:SetFrequency(freq)
	forward,backward = S:GetPoyntingFlux('AirAbove', 0)
	forward = S:GetPoyntingFlux('AirBelow', 0)
	print (freq .. '\t' .. forward .. '\t' .. backward)
	io.stdout:flush()
end

File: examples/Fan_PRB_65_2002/fig12.R
# Bottom pane of Fig. 12 in
# Shanhui Fan and J. D. Joannopoulos,
# "Analysis of guided resonances in photonic crystal slabs",
# Phys. Rev. B, Vol. 65, 235112

require("RCWA")
S <- new(Simulation,matrix(c(1,0,0,1),nrow=2,ncol=2), 100)

S$set_material("Vacuum", 1)
S$set_material("Silicon", 12 + 0i)

S$add_layer("AirAbove", 0 , "Vacuum")
S$add_layer("Slab", 0.5, "Silicon")
S$pattern_circle("Slab", "Vacuum", c(0,0), 0.2)
S$add_layer("AirBelow", 0, "Vacuum")

S$excitation_planewave(
	c(0,0), # incidence angles (radians)
	1, # s-polarization (complex amplitude)
	0) # p-polarization (complex amplitude)

for (freq in seq(0.25, 0.6, length.out=100)){
	S$set_frequency(freq)
	powers_top = S$get_power_flux("AirAbove", 0)
	powers_bot = S$get_power_flux("AirBelow", 0)
	cat(sprintf("%f\t%f\t%f\n", freq, powers_bot[["forward"]], powers_top[["backward"]]))
}
